templates/deno/package.json

```
@@ -16,5 +16,11 @@
   },
   "engines": {
     "node": ">=14"
+  },
+  "dependencies": {
+    "@remix-run/deno": "*",
+    "@remix-run/react": "*",
+    "react": "^18.1.0",
+    "react-dom": "^18.1.0"
```

I think we should keep using v17 for now, so we're in line with all other templates.

Shouldn't we add `@types/react`, `@types/react-dom` & `typescript` in `devDependencies` too?

```suggestion
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
```

---

We don't need `@types/*` nor `typescript` in `devDependencies` because users will be relying on Deno tools (like the VSCode Deno extension) to provide types. These tools will use import maps to find the appropriate types (e.g. `templates/deno/.vscode/resolve_npm_imports.json`).